### PR TITLE
docs: feature-freeze unmaintained components

### PR DESCRIPTION
**Pull request by pichlermarc**
We often get feature PRs and `feature-request` issues and feature code-contributions opened for components that are in the `Unmaintained` state, and therefore do not have any owners/reviewers. The SIG does not have the capacity to review these one-off feature-contributions for unmaintained components, and it is currently not clear to contributors that this is the case due to a lack of documentation on that topic.

This PR 
- proposes moving all `Unmaintained` components to a feature-freeze state
  - features may only be added when there's a Sponsor for that feature (or a new Owner has been found). Bugs and semconv updates are exempt from this limitation.
- PRs for `Unmaintained` components will be auto-labelled with `pkg-status:unmaintained`
- Issues for `Unmaintained` components have to be labelled manually with `pkg-status:unmaintained`
- adds a stale step that will close feature issues and PRs for unmaintained components every morning if they've been open for 14 days.
  - this step will also let people know what the `pkg-status:unmaintained` label means for the issue or PR
  - `has:sponsor` will exempt issues and PRs from being auto-closed
  - `type:semconv-update` will exempt issue and PR from being auto-closed
  - `bug` will exempt issue and PR from being auto-closed

### CODE REVIEW

File: .github/workflows/close-stale.yml
Comment (dyladan) [inline .github/workflows/close-stale.yml:35]:
```suggestion
          stale-pr-message: 'This package does not have an assigned owner and is considered unmaintained. As such this package is in feature-freeze and this PR will be closed with 14 days unless a new owner or a sponsor (@open-telemetry/javascript-approvers) for the feature is found. It is the responsibility of the author to find a sponsor for this PR.'
```

Comment (pichlermarc) [reply .github/workflows/close-stale.yml]: Applied in https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2272/commits/58d00e74251fa582c9da4ec303e5b2aef1a5baf6

**Comment by pichlermarc** [comment]
@dyladan https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2272/commits/58d00e74251fa582c9da4ec303e5b2aef1a5baf6 adds the exemption for semconv updates we talked about yesterday. :slightly_smiling_face:

File: .github/workflows/close-stale.yml
Comment (maryliag) [inline .github/workflows/close-stale.yml:27]: do you want to add something for "have you consider becoming a code owner of this package?"

File: CONTRIBUTING.md
Comment (maryliag) [inline CONTRIBUTING.md:198]: We also mentioned that we would allow bug fixes, do you want to clarify this here too?

File: .github/workflows/close-stale.yml
Comment (pichlermarc) [reply .github/workflows/close-stale.yml:27]: Hmm, I think this is a good idea. Actually, we'd need some description of "how to become a component owner" which is currently missing. I'll open another PR for this as I think it's a separate topic that we should link to there.

Comment (pichlermarc) [reply .github/workflows/close-stale.yml:27]: See #2287. Once merged, I'll add a link here so people know what to do to become a component owner.

File: CONTRIBUTING.md
Comment (pichlermarc) [reply CONTRIBUTING.md]: Good point, clarified that in https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2272/commits/655740968e0af8276c0554399a6d8e2d8d266e84 :slightly_smiling_face:

File: .github/workflows/close-stale.yml
Comment (maryliag) [reply .github/workflows/close-stale.yml:27]: Great! Once you add the link, I think this Pr is good to get approved 😄

**Review by blumamir — APPROVED** [review]
I support linking to the relevant part of the CONTRIBUTING.md. 
other than that, LGTM

**Comment by pichlermarc** [comment]
@maryliag @blumamir I added some links to the message that the action will post which links to the `CONTRIBUTING.md` which explain the concepts (see https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2272/commits/cbfd085644df46bbd26fd70af2c02c4f2b3cce82)

File: .github/workflows/close-stale.yml
Comment (pichlermarc) [reply .github/workflows/close-stale.yml]: Added the links (what unmaintained means, and how to become an owner): https://github.com/open-telemetry/opentelemetry-js-contrib/commit/cbfd085644df46bbd26fd70af2c02c4f2b3cce82 :slightly_smiling_face:

**Review by maryliag — APPROVED** [review]

**Review by legendecas — APPROVED** [review]

**Pull request merged**
